### PR TITLE
vxlan: T2629: fix multiple configuration issues

### PR DESCRIPTION
**Pull request by SIN3R6Y**
https://phabricator.vyos.net/T2629

The configuration option "source-interface" maps to "src_interface" on the command line which is invalid, it should map to "dev".

The configuration option "source-address" forces the "nolearning" flag, which will break most switches with hardware vtep features. If we want to offer this flag, it should have it's own configuration.

Setting "remote" and "source-address" results in only "remote" being applied, even though "source-address" saves to the configuration.

Erroring out on "group" / "src_interfaces" should only happen when one of them is configured, not when the general configuration is incomplete. Otherwise the error message is rather cryptic.

### CODE REVIEW

**Comment by c-po** [comment]
@thomas-mangin @sever-sever can you review this please?

File: python/vyos/ifconfig/vxlan.py [hunk 1]
```diff
@@ -66,24 +66,26 @@ class VXLANIf(Interface):
...
+            cmdline.append('src_address')
 
-        else:
-            ifname = self.config['ifname']
-            raise ConfigError(
-                f'VXLAN "{ifname}" is missing mandatory underlay interface for a multicast network.')
+        if self.config['remote']:
+            cmdline.append('remote')
+
+        if self.config['group'] or self.config['src_interface']:
```
Comment (thomas-mangin) [inline python/vyos/ifconfig/vxlan.py:82]: Could this be better written as ?

```
if self.config['group'] and self.config['src_interface']:
    cmdline.append('group', 'src_interface')
elif self.config['group'] or self.config['src_interface']:
     ifname = self.config['ifname']
     raise ConfigError(
                     f'VXLAN "{ifname}" is missing mandatory underlay multicast group or source interface for a multicast network.') 
```

**Comment by thomas-mangin** [comment]
One suggestion but can not see any problem, the patch remove the no learning so probably a companion patch and migration code may be required to allow the behaviour and keep backward behaviour. Can not make a call on that point.

**Comment by SIN3R6Y** [comment]
> One suggestion but can not see any problem, the patch remove the no learning so probably a companion patch and migration code may be required to allow the behaviour and keep backward behaviour. Can not make a call on that point.

In my opinion, removing nolearning has no negative side effects. It's not explicit mentioned anywhere that nolearning was enabled, and it's in my opinion a bad assumption that it should be enabled automatically in the first place.

all "nolearning" does is prevent the interface from sending learning packets to the remote end of the tunnel. For point to point tunnels with two linux based hosts, this works fine. But with other network devices they expect to see the learning packets in order to establish the tunnel.

removing nolearning still works just fine on existing systems, and allow vxlans to communicate with other systems aside from linux, the only cost is an extra few packets here and there. For what it's worth, the linux default is to leave learning enabled.

**Comment by sever-sever** [comment]
> @thomas-mangin @sever-sever can you review this please?
@c-po I checked with this configuration and it worked
```
set interfaces vxlan vxlan14 source-address '100.64.0.1'
set interfaces vxlan vxlan14 vni '14'

```

**Comment by sever-sever** [comment]
"No learning"  is needed for this instruction.
https://vincent.bernat.ch/en/blog/2017-vxlan-bgp-evpn

**Comment by c-po** [comment]
@SIN3R6Y the `nolearning` stuff was adde dn 1.3 for a VXLAN EVPN test PoC. It's not yet "production" code nor is EVPN integrated in VyOS at a CLI level so we can shift things around here.

@sever-sever probably best is if you approve this or not ;)

**Comment by sever-sever** [comment]
@c-po I approve

**Pull request merged**